src/main/java/org/prebid/server/bidder/kubient/KubientBidder.java

```diff
@@ -0,0 +1,117 @@
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * kubient {@link Bidder} implementation.
+ */
+public class KubientBidder implements Bidder<BidRequest> {
+
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+
+    public KubientBidder(String endpointUrl) {
+        this.endpointUrl = HttpUtil.validateUrl(endpointUrl);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+
+        String body;
+        try {
+            body = Json.encode(request);
+        } catch (EncodeException e) {
+            errors.add(BidderError.badInput(String.format("Failed to encode request body, error: %s", e.getMessage())));
+            return Result.of(Collections.emptyList(), errors);
+        }
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .body(body)
+                        .headers(HttpUtil.headers())
+                        .payload(request)
+                        .build()),
+                errors);
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        if (httpCall.getResponse().getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+
+        try {
+            final BidResponse bidResponse = Json.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return extractBids(httpCall.getRequest().getPayload(), bidResponse);
+        } catch (DecodeException | PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private Result<List<BidderBid>> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+        final List<BidderBid> bidderBids = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+        for (SeatBid seatBid : bidResponse.getSeatbid()) {
+            if (seatBid != null) {
+                final List<Bid> bids = seatBid.getBid();
+                if (bids != null) {
+                    for (Bid bid : bids) {
+                        try {
+                            final BidType bidType = getBidType(bid.getImpid(), bidRequest.getImp());
+                            bidderBids.add(BidderBid.of(bid, bidType, DEFAULT_BID_CURRENCY));
+                        } catch (PreBidException e) {
```

Comment:
FYI, this is fine, but just to give you a possible work-around for when you need to catch errors in `Stream` - pass an errors list to bid mapper function and when an error occurs - add it to list, but return `null` instead. Then just filter for non-null, like this:

```
bidResponse.getSeatbid().stream()
                .map(SeatBid::getBid)
                .filter(Objects::nonNull)
                .flatMap(Collection::stream)
                .map(bid -> toBidderBid(bid, imps, errors))
                .filter(Objects::nonNull)
                .collect(Collectors.toList())
```

Reply:
I noticed that variant also. I needed to make a decision between pass and modifying collection in another method and smelly code.

I'll change to `stream` variant